processors/src/main/java/io/opentelemetry/contrib/interceptor/InterceptableLogRecordExporter.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.interceptor;
+
+import io.opentelemetry.contrib.interceptor.api.Interceptor;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.logs.data.LogRecordData;
+import io.opentelemetry.sdk.logs.export.LogRecordExporter;
+import java.util.Collection;
+
+/** Intercepts logs before delegating them to the real exporter. */
+public class InterceptableLogRecordExporter implements LogRecordExporter {
```

Comment:
Nit: all of these exporters can be `final`
```suggestion
public final class InterceptableLogRecordExporter implements LogRecordExporter {
```